connectivity/service/src/main/java/org/eclipse/ditto/connectivity/service/mapping/CloudEventsMapper.java

```diff
@@ -0,0 +1,166 @@
+    })
+public final class CloudEventsMapper extends AbstractMessageMapper {
+
+  static final String CE_ID = "ce-id";
+  static final String CE_TYPE = "ce-type";
+  static final String CE_SOURCE = "ce-source";
+  static final String CE_TIME = "ce-time";
+
+  static final JsonObject DEFAULT_OPTIONS =
+      JsonObject.newBuilder()
+          .set(
+              MessageMapperConfiguration.CONTENT_TYPE_BLOCKLIST,
+              String.join(
+                  ",",
+                  "application/vnd.eclipse-hono-empty-notification",
+                  "application/vnd.eclipse-hono-device-provisioning-notification",
+                  "application/vnd.eclipse-hono-dc-notification+json",
+                  "application/vnd.eclipse-hono-delivery-failure-notification+json"))
+          .build();
+
+  /** The context representing this mapper */
+  public static final MappingContext CONTEXT =
+      ConnectivityModelFactory.newMappingContextBuilder(
+              CloudEventsMapper.class.getCanonicalName(), DEFAULT_OPTIONS)
+          .build();
+
+  final String SPECVERSION = "specversion";
+  final String ID = "id";
+  final String SOURCE = "source";
+  final String TYPE = "type";
+
+  public String base64decoding(final String base64Message) throws UnsupportedEncodingException {
+    byte[] messageByte = Base64.getDecoder().decode(base64Message);
+    String decodedString = new String(messageByte);
+    return decodedString;
+  }
+
+  boolean validatePayload(String payload) {
+    Map<String, Object> incomingMessagePayload = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+    final JsonObject jsonObject = JsonFactory.newObject(payload);
+    List<JsonKey> key = jsonObject.getKeys();
+    for (JsonKey keyIterator : key) {
+      Object value = jsonObject.getValue(keyIterator).orElse(null);
+      if (value != null) {
+        incomingMessagePayload.put(keyIterator.toString(), value);
+      }
+    }
+    if (incomingMessagePayload.get("specversion") == null
+        || incomingMessagePayload.get("id") == null
+        || incomingMessagePayload.get("source") == null
+        || incomingMessagePayload.get("type") == null) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  private JsonifiableAdaptable extractData(final String message)
+      throws UnsupportedEncodingException {
+    final JsonObject payloadJson = JsonFactory.newObject(message);
+    if (payloadJson.getValue("data_base64").isPresent()) {
+      String base64Data = payloadJson.getValue("data_base64").orElse(null).asString();
+      String decodedData = base64decoding(base64Data);
+
+      final JsonifiableAdaptable decodedJsonifiableAdaptable =
+          DittoJsonException.wrapJsonRuntimeException(
+              () ->
+                  ProtocolFactory.jsonifiableAdaptableFromJson(JsonFactory.newObject(decodedData)));
+      return decodedJsonifiableAdaptable;
+
+    } else {
+      String data = payloadJson.getValue("data").get().toString();
```

Comment:
when `"data"` is not in the JSON, this will throw an exception.
Please handle that more explicitly (e.g. as another "if" cause).